UnityProject/Assets/Scripts/Items/Weapons/ExplosionUtils.cs

```diff
@@ -26,20 +23,22 @@ public static class ExplosionUtils
 		/// <param name="shakeDistance">how far away the shaking can be felt</param>
 		public static void PlaySoundAndShake(Vector3Int worldPosition, byte shakeIntensity, int shakeDistance)
 		{
-			AddressableAudioSource sndName = EXPLOSION_SOUNDS[Random.Range(0, EXPLOSION_SOUNDS.Length)];
+
+			AddressableAudioSource explosionSound = EXPLOSION_SOUNDS.PickRandom();
+			AddressableAudioSource groanSound     = STATION_GROAN_SOUNDS.PickRandom();
+			AddressableAudioSource distantSound   = DISTANT_EXPLOSION_SOUNDS.PickRandom();
 			AudioSourceParameters audioSourceParameters = new AudioSourceParameters(0f, 100f);
 			ShakeParameters shakeParameters = new ShakeParameters(true, shakeIntensity, shakeDistance);
-			_ = SoundManager.PlayNetworkedAtPosAsync(sndName, worldPosition, audioSourceParameters, true, false, shakeParameters: shakeParameters);
 
-			if (shakeDistance > DISTANT_THRESHOLD)
+			_ = SoundManager.PlayNetworkedAtPosAsync(explosionSound, worldPosition, audioSourceParameters, true, false, shakeParameters);
+
+			if (shakeDistance > distantSound.AudioSource.maxDistance)
 			{
-				AddressableAudioSource distantName = DISTANT_EXPLOSION_SOUNDS[Random.Range(0, DISTANT_EXPLOSION_SOUNDS.Length)];
-				_ = SoundManager.PlayNetworkedAtPosAsync(distantName, worldPosition, global: true);
+				_ = SoundManager.PlayNetworkedAtPosAsync(distantSound, worldPosition, global: true);
 			}
-			if (shakeIntensity > GROAN_THRESHOLD)
+			if (shakeIntensity > groanSound.AudioSource.maxDistance)
 			{
-				AddressableAudioSource groanName = STATION_GROAN_SOUNDS[Random.Range(0, STATION_GROAN_SOUNDS.Length)];
```

Comment:
wouldn't this still play across matrices?

Reply:
hm? wdym?

Reply:
if you're on an asteroid You will still hear this because it's global

Reply:
Daniel explained this ``global`` variable is just misnamed. ``global`` in this context means "everyone spawns the prefab" but the falloff distance set in the prefab still applies.

Reply:
ok cool